src/QoiSharp/QoiImage.cs

```diff
@@ -31,13 +35,19 @@ public class QoiImage
     /// Color space.
     /// </summary>
     public ColorSpace ColorSpace { get; }
-    
+
     /// <summary>
     /// Default constructor.
     /// </summary>
     public QoiImage(byte[] data, int width, int height, Channels channels, ColorSpace colorSpace = ColorSpace.SRgb)
+        : this(new ReadOnlyMemory<byte>((byte[])data.Clone()), width, height, channels, colorSpace)
```

Comment:
Ensures that if someone passes in a byte[], further mutations to that buffer will not corrupt the `QoiImage` object.